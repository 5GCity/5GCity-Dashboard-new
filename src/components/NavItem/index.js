/**
 * NavItem Component
 * Please write a description
 *
 * @author Guilherme Patriarca <gpatriarca@ubiwhere.com>
 */
import React from 'react'
import styled from 'styled-components'
import { Icon } from 'element-react'
import {  } from 'polished'
/* import PropTypes from 'prop-types' */


export default ({ children, icon, text, disabled, href, onClick, width, active, ...props }) => (
      <Wrapper
        active={active}
        disabled={disabled}
        onClick={!disabled ? onClick : null}
        width={width}
        to={href}
      >
        <IconItem name={icon}></IconItem>
          <ItemLabel>{text}</ItemLabel>
      </Wrapper>
    )

const Wrapper = styled.div`
  color: #fff;
  text-align: center;
  cursor: pointer;
  height: 92px;
  padding: 14px 16px;
  position: relative;
  box-shadow: inset 0 -1px 0 0 rgba(137,151,159,0.15);


   :hover {
    background-color: rgba(137,151,159,0.2);
  }

  ${({ active }) => active && `
    border-right: 3px solid #8CC14E;
    :after {
      content:'';
      position: absolute;
      border: 30px solid transparent;
      border-right: 30px solid #8CC14E;
      top: 40px;
      right: 0px;
    }
`}


`

const ItemLabel = styled.div`
  line-height: 15px;
  margin-top: 8px;
  font-size: 14px;
  text-transform: uppercase;
  font-family: ${({ theme }) => theme.secondaryFont};
`

const IconItem = styled(Icon)`
  font-size:32px;
`