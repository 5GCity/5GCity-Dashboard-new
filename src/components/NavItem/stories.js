/**
 * NavItem Component Stories
 * Please write a description or remove this line
 *
 * @author Your Name <youremail@ubiwhere.com>
 */
import React from 'react'
import { storiesOf } from '@storybook/react'
import styled from 'styled-components'
import { Theme } from 'globalStyles'

import NavItem from './index'

const bStyle = {
  padding:30,
  backgroundColor: Theme.bodyBackground,
  height:'100%'
};



const exampleOne = () => (
  <div style={bStyle}>
    <NavItem
      active={true}
      icon={'edit'}
      text={'Item 1'}
      disabled={false}
      href={''}
      onClick={() => console.log("ola")}
    />
    <NavItem
      active={true}
      icon={'view'}
      text={'Item 2'}
      disabled={false}
      href={''}
    />
  </div>
)

storiesOf('NavItem', module)
  .add('primary', exampleOne)