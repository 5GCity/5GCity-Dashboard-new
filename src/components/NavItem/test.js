/**
 * NavItem Component Tests
 *
 * @author Your Name <youremail@ubiwhere.com>
 */
import React from 'react'
import renderer from 'react-test-renderer'
import NavItem from './index'
import 'jest-styled-components'

it('renders correctly', () => {
  const tree = renderer
    .create(<NavItem />)
    .toJSON()

  expect(tree).toMatchSnapshot()
})
