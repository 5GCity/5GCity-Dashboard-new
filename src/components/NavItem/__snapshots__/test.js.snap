// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c0 {
  color: #fff;
  text-align: center;
  cursor: pointer;
  height: 92px;
  padding: 14px 16px;
  position: relative;
  box-shadow: inset 0 -1px 0 0 rgba(137,151,159,0.15);
}

.c0:hover {
  background-color: rgba(137,151,159,0.2);
}

.c1 {
  font-size: 32px;
}

.c2 {
  line-height: 15px;
  margin-top: 8px;
  font-size: 14px;
  text-transform: uppercase;
}

<div
  className="c0"
>
  <i
    className="el-icon-undefined c1"
    style={Object {}}
  />
  <div
    className="c2"
  />
</div>
`;
